### PR TITLE
feature(devservices): added firestore dev service

### PR DESCRIPTION
**Pull request by zZHorizonZz**
This PR includes the Firestore dev service. Additionally, it renames Pub/Sub to just PubSub in the documentation and simplifies some variable names in the PubSub dev service processor.

### CODE REVIEW

**Comment by loicmathieu** [comment]
@zZHorizonZz The official name of the product is Pub/Sub so I prefer to keep it https://cloud.google.com/pubsub

As a side note, I prefer to have separate PR when there is two things changed, easier to review it ;) any way big thank you for doing it.

**Comment by zZHorizonZz** [comment]
> @zZHorizonZz The official name of the product is Pub/Sub so I prefer to keep it https://cloud.google.com/pubsub

Thanks for the info I will revert it. 👍

File: firestore/deployment/src/main/java/io/quarkiverse/googlecloudservices/firestore/deployment/FirestoreDevServiceConfig.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+     */
+    @ConfigItem(defaultValue = "false")
+    public boolean enabled;
+
+    /**
+     * Sets the Docker image name for the Google Cloud SDK.
+     * This image is used to emulate the Firestore service in the development environment.
+     * The default value is 'gcr.io/google.com/cloudsdktool/google-cloud-cli'.
+     */
+    @ConfigItem(name = "image-name", defaultValue = "gcr.io/google.com/cloudsdktool/google-cloud-cli")
```
Comment (loicmathieu) [inline firestore/deployment/src/main/java/io/quarkiverse/googlecloudservices/firestore/deployment/FirestoreDevServiceConfig.java:36]: I think the name is not mandatory here, this is the default naming translation.

File: firestore/deployment/src/main/java/io/quarkiverse/googlecloudservices/firestore/deployment/FirestoreDevServiceConfig.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    public String imageName;
+
+    /**
+     * Specifies the emulatorPort on which the Firestore service should run in the development environment.
+     */
+    @ConfigItem(name = "emulatorPort")
```
Comment (loicmathieu) [inline firestore/deployment/src/main/java/io/quarkiverse/googlecloudservices/firestore/deployment/FirestoreDevServiceConfig.java:42]: For simplicity purpose, I prefer not using name but name the attribute with the right name

File: firestore/deployment/src/main/java/io/quarkiverse/googlecloudservices/firestore/deployment/FirestoreDevServiceProcessor.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        timeout.ifPresent(emulatorContainer::withStartupTimeout);
+        emulatorContainer.start();
+
+        // Set the config for the started container
+        FirestoreDevServiceProcessor.config = config;
+
+        // Return running service item with container details
+        return new DevServicesResultBuildItem.RunningDevService(FirestoreBuildSteps.FEATURE,
+                emulatorContainer.getContainerId(),
+                emulatorContainer::close, "quarkus.google.cloud.firestore.emulator-host",
```
Comment (loicmathieu) [inline firestore/deployment/src/main/java/io/quarkiverse/googlecloudservices/firestore/deployment/FirestoreDevServiceProcessor.java:124]:
```suggestion
                emulatorContainer::close, "quarkus.google.cloud.firestore.host-override",
```

There is an existing property for that.

File: firestore/runtime/src/main/java/io/quarkiverse/googlecloudservices/firestore/runtime/FirestoreProducer.java [hunk 4]
```diff
@@ -8,6 +8,7 @@
 import jakarta.inject.Inject;
 import jakarta.inject.Singleton;
 
+import org.eclipse.microprofile.config.inject.ConfigProperty;
```
Comment (loicmathieu) [inline firestore/runtime/src/main/java/io/quarkiverse/googlecloudservices/firestore/runtime/FirestoreProducer.java:11]: All the changes on this class should not be needed if using the default quarkus.google.cloud.firestore.host-override property.

**Review by loicmathieu — CHANGES_REQUESTED** [review]

**Review by loicmathieu — APPROVED** [review]
LGTM, thanks a lot

**Pull request merged**